### ISSUE TITLE
Floating Point Attacks

### ISSUE THREAD

**Issue by messerman** [michaelchadwick/gemwarrior#35]
Attacking a CUBICAT in the arena, my attacks were (often) hitting for non-integer values (which also take a long time to display). Not necessarily a bug (kind of funny, actually), but then the HP needs to be displayed appropriate, as well, as you can see I hit for 1.2ish damage, but CUBICAT failed to die.

---

LANEN        ::  17 HP
CUBICAT      ::   1 HP

What do you do?
[Fight/Attack][Defend][Look][Pass][Run]
a
You attack cubicat with your gun!

> You wound cubicat for 0.6781066586209497 point(s)!
> cubicat attacks you!
> You are wounded for 14 point(s)!
> You are almost dead!

cubicat is almost dead!

LANEN        ::   3 HP
CUBICAT      ::   1 HP

What do you do?
[Fight/Attack][Defend][Look][Pass][Run]
a
You attack cubicat with your gun!

> You wound cubicat for 0.5609585703132399 point(s)!
> cubicat attacks you!
> You are wounded for 13 point(s)!
> You are dead, slain by the cubicat!
> ---

**Comment by michaelchadwick** [comment]
This is an interesting bug, and I can't seem to reproduce it, either with the gun or without, in the arena or not. I've never seen decimal damage points thus far! If you can tell me how to reproduce it, I'll try to fix it.

**Comment by messerman** [comment]
It only happened in the one fight. If you want, I'll look through the code for this (and also rest), see if anything jumps out.

## Matt

iPhone'd

**Comment by michaelchadwick** [comment]
That's the allure of Github, isn't it? ;P

**Comment by messerman** [comment]
The issue is in the following code, if the player has the ability of rock_slide, BUT they don't hit it, then atk_range is never set (which should actually cause an exception, unless it's initialized somewhere else, but it's not as far as I can tell).

``` ruby
          if player.beast_mode
            atk_range = BEAST_MODE_ATTACK..BEAST_MODE_ATTACK
          elsif player.special_abilities.include?(:rock_slide)
            lo_boost = rand(0..9)

            if lo_boost >= 7
              puts "#{player.name} uses Rock Slide for added damage!"
              hi_boost = lo_boost + rand(0..5)
              atk_range = (player.atk_lo + lo_boost)..(player.atk_hi + hi_boost)
            end
          else
            atk_range = player.atk_lo..player.atk_hi
          end
          rand(atk_range)
```

Try this, instead:

``` ruby
          atk_range = player.atk_lo..player.atk_hi

          if player.beast_mode
            atk_range = BEAST_MODE_ATTACK..BEAST_MODE_ATTACK
          elsif player.special_abilities.include?(:rock_slide)
            lo_boost = rand(0..9)
            if lo_boost >= 7
              puts "#{player.name} uses Rock Slide for added damage!"
              hi_boost = lo_boost + rand(0..5)
              atk_range = (player.atk_lo + lo_boost)..(player.atk_hi + hi_boost)
            end
          end

          rand(atk_range)
```

**Comment by michaelchadwick** [comment]
This is the answer. Thanks!

**Comment by michaelchadwick** [comment]
Fixed in latest release (0.9.23)
